tests/model/layer.test.js

```diff
@@ -1,71 +1,137 @@
+
+var connection1 = new Connection('conn-1-name', 'unique-id-conn-1', sourceTerm1, [sinkTerm1],
+        [segment1, segment2]);
+var connection2 = new Connection('conn-2-name', 'unique-id-conn-2', sourceTerm2, [sinkTerm2],
+        [segment3, segment4]);
 
 describe('initialization', () => {
-    test('constructor', () => {
-        let layer = new Layer('layer-name', 'layer-id');
+    describe('constructor', () => {
+        test('parameters', () => {
+            let layer = new Layer('layer-name', 'layer-id', [component1, component2], [connection1, connection2]);
+
+            expect(layer.name).toBe('layer-name');
+            expect(layer.id).toBe('layer-id');
+            expect(layer.components).toEqual([component1, component2]);
+            expect(layer.connections).toEqual([connection1, connection2]);
+        });
 
-        expect(layer.name).toBe('layer-name');
-        expect(layer.id).toBe('layer-id');
+        test('default', () => {
+            let l = new Layer();
+
+            expect(l.name).toBe(Validation.DEFAULT_STR_VALUE);
+            expect(l.id).toBe(Validation.DEFAULT_STR_VALUE);
+            expect(l.components).toEqual([]);
+            expect(l.connections).toEqual([]);
+        });
     });
 
     test('modify fields', () => {
         let layer = new Layer('layer-name', 'layer-id');
 
         layer.name = 'new-layer-name';
         layer.id = 'new-layer-id';
+        layer.components = [component1, component2];
+        layer.connections = [connection1, connection2];
 
         expect(layer.name).toBe('new-layer-name');
         expect(layer.id).toBe('new-layer-id');
+        expect(layer.components).toEqual([component1, component2]);
+        expect(layer.connections).toEqual([connection1, connection2]);
     });
 });
 
 describe('validation', () => {
     describe('invalid', () => {
-        test(' name value', () => {
-            let badNameLayer = new Layer('', 'id');
+        test('all fields', () => {
+            let badBothLayer = new Layer(Validation.DEFAULT_STR_VALUE, Validation.DEFAULT_STR_VALUE);
+
+            expect(badBothLayer.validate()).toBe(false);
+        });
+
+        test('name', () => {
+            let badNameLayer = new Layer(Validation.DEFAULT_STR_VALUE, 'id', [component1, component2],
+                    [connection1, connection2]);
 
             expect(badNameLayer.validate()).toBe(false);
         });
 
-        test('id value', () => {
-            let badIDLayer = new Layer('name', '');
+        test('id', () => {
+            let badIDLayer = new Layer('name', Validation.DEFAULT_STR_VALUE, [component1, component2],
+                    [connection1, connection2]);
 
             expect(badIDLayer.validate()).toBe(false);
         });
 
-        test('name and id values', () => {
-            let badBothLayer = new Layer('', '');
+        test('components', () => {
+            let l = new Layer('name', 'id', [new Component(), component2], [connection1, connection2]);
```

Comment:
The switch from the seemingly established usage of `goodLayer` and `badLayer` to this shortened `l` seems strange here to me.

Reply:
Because the tests are more atomic, the ```describe``` and ```test``` calls describe what the object is. The ```goodLayer``` and ```badLayer``` was left over from when I was testing a lot of different things together. I've changed the shortened ones to ```bad*``` keep with the convention.